### DISCUSSION TITLE
Problem with MOA of subgrain boundaries

### DISCUSSION THREAD

**Discussion by rockdef** [mtex-toolbox/mtex#1975] [Ask Anything]
Hi, 

Another issue arises when plotting the MOA of subgrain boundaries. 

**Error**: There is no indexed data in this variable! (I can see that there is no data on subgrain boundaries in the workspace). Somewhere in the code, I am deleting them, and I'm unable to trace where it's happening.

```matlab
%% Specify Crystal and Specimen Symmetries
CS = {... 
  'notIndexed',...
  crystalSymmetry('12/m1', [9.9 18 5.3], [90,105.2,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Hornblende', 'color', [1 0.6 0])};

%% Miller Indices
h_hbl = [Miller(1,0,0,CS{2},'hkl'), Miller(0,1,0,CS{2},'hkl'),Miller(0,0,1,CS{2},'hkl'), Miller(1,1,0,CS{2},'hkl')];

%% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Import the Data
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertEuler2SpatialReferenceFrame','setting 2');

%% Rotation 
ebsd = rotate(ebsd,rotation('axis',vector3d.X,'angle',180*degree));

%% Compute preliminary grain boundaries
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',[10 1]*degree,'Boundary','tight');

%% Inverse Pole Figures (IPF)

gB_hbl = grains.boundary('Hornblende','Hornblende'); %
ind1 = angle(gB_hbl.misorientation)<10*degree;
lowangleBoundary = gB_hbl(ind1) ;
gBsub=lowangleBoundary;
gBinn=grains.innerBoundary('Hornblende','Hornblende');

ng_SB = length(gBsub); %number of grains
ng_IB = length(gBinn); %number of grains

% IPF of High-angle grain boundary (HAGB) (> 10)
figure,
HAGB=grains_smooth.boundary('Hornblende','Hornblende');
plotAxisDistribution(HAGB.misorientation,'contourf')
title('High-angle grain boundaries','fontsize',14)
mtexColorbar('title','M.U.D.');

% IPF of Subgrain Boundaries
figure,
plotAxisDistribution(gBsub.misorientation,'contourf')
title('Subgrain boundaries','fontsize',14)
ngrains = "N = " + ng_SB; % Variable for box 2
dim1 = [0.02 0.48 0.5 0.5]; % Top-left corner
annotation('textbox',dim1,'String',ngrains,'FitBoxToText','on','fontsize',12); % Box with grains number
mtexColorbar('title','M.U.D.');

% IPF of Inner Boundaries
figure,
plotAxisDistribution(gBinn.misorientation,'contourf')
title('Inner boundaries','fontsize',14)
ngrains = "N = " + ng_IB; % Variable for box 2
dim2 = [0.02 0.48 0.5 0.5]; % Top-left corner
annotation('textbox',dim2,'String',ngrains,'FitBoxToText','on','fontsize',12); % Box with grains number
mtexColorbar('title','M.U.D.');
```
**NOTE:** 
I tried to use this function as per the suggestion. 
condLAB = sub.misorientation.angle>=2*degree & sub.misorientation.angle<10*degree;
But Mtex says - Unable to resolve the name 'sub.misorientation.angle'.

Does anyone have any idea how to encounter this problem? 
Thanks in advance!

**Comment by zmichels** [comment]
Hi,

It looks like you use a 10-degree minimum for grain boundaries. Then, for gBsub, you query which of those boundaries have <10-degree misorientations, from which to extract axes. However, I wouldn't expect there to be any boundaries in that variable given the threshold in reconstruction. Maybe what you want to use is grains.innerBoundary instead of grains.boundary? This is what you did for 'gBinn' variable... are there data in this variable?

**Reply by rockdef** [reply]
Hi Zach, 

Sorry for the late response due to the difficult times here. 

Sorry, I didn't understand completely your questions.
I wanted to plot three misorientation axes (MOA) in the inverse pole figure (IPF): 
1. Misorientation axis (MOA) for high-angle grain boundaries (HAGB) (> 10*degree) 
2. Misorientation axis (MOA) for low-angle subgrain boundaries (LASB) (between 2*degree - 10*degree) 
3. Misorientation axis (MOA) for low-angle inner boundaries (LAIB) (misorientation > 2*degree) 

For this, I don't think my command is wrong. 
I am not sure why it's showing 'No index data'. 

Thank you very much in advance!

**Reply by zmichels** [reply]
When you compute grains, the resulting structure contains "boundary" and "innerBoundary". The result of grains.boundary will return boundaries that have misorientations >= the highest-angle threshold you specified in calcGrains(). I believe you used 10°. So all boundaries from grains.boundary should have misorientations of >=10°. If you take that set of boundaries and query which of them have boundaries between 2-10° (less than 10°), I would expect that you would get an empty set as a result.

If you prefer not to use innerBoundaries, or feel the need to do something additional... you could try computing grains a second time using a much lower angle threshold for the outer boundaries, and then subset those.

Does that make sense?
